akka-stream-tests/src/test/scala/akka/stream/io/InputStreamSourceSpec.scala

```diff
@@ -21,6 +22,17 @@ class InputStreamSourceSpec extends AkkaSpec(UnboundedMailboxConfig) with ScalaF
   implicit val materializer = ActorMaterializer(settings)
 
   "InputStreamSource" must {
+
+    "not signal when no demand" in {
+      val f = StreamConverters.fromInputStream(() ⇒ new InputStream {
+        override def read(): Int = 42
+      })
+
+      Await.result(f
+        .takeWithin(5.seconds)
+        .runForeach(it ⇒ ()), 10.seconds)
+    }
```

Comment:
not really, this tests that it did not blow up – during the 5 seconds it was operating